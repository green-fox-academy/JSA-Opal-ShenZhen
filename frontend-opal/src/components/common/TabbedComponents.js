import React from 'react';
import { Tab, Tabs } from 'native-base';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  tabs: {
    zIndex: 10
  },
  tabBarUnderlineStyle: {
    backgroundColor: 'rgb(250,110,59)'
  },
  tabStyle: {
    backgroundColor: 'white'
  },
  textStyle: {
    color: 'rgb(250,110,59)'
  },
  activeTabStyle: {
    backgroundColor: 'white'
  }
});

function TabbedComponents({ components }) {
  return (
    <Tabs style={styles.tabs} tabBarUnderlineStyle={styles.tabBarUnderlineStyle}>
      {components.map(Component => (
        <Tab
          heading={Component.title}
          key={Component.title}
          tabStyle={styles.tabStyle}
          textStyle={styles.textStyle}
          activeTabStyle={styles.activeTabStyle}
          activeTextStyle={styles.textStyle}
        >
          <Component />
        </Tab>
      ))}
    </Tabs>
  );
}

TabbedComponents.propTypes = {
  components: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default TabbedComponents;
