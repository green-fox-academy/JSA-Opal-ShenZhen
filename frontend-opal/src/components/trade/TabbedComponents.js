import React, { useEffect, useContext } from 'react';
import { Tab, Tabs } from 'native-base';
import { StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { NavigationContext } from 'react-navigation';
import PropTypes from 'prop-types';

import BuyContainer from 'components/trade/BuyContainer';
import SellContainer from 'components/trade/SellContainer';
import thunks from 'thunks/instrumentDetail';
import actions from 'actions/instrumentDetail';
import presetProps from 'components/InstrumentDetail/presetProps';

const styles = StyleSheet.create({
  tabs: {
    zIndex: 10
  },
  tabBarUnderlineStyle: {
    backgroundColor: 'rgb(250,110,59)'
  },
  tabStyle: {
    backgroundColor: 'white'
  },
  textStyle: {
    color: 'rgb(250,110,59)'
  },
  activeTabStyle: {
    backgroundColor: 'white'
  }
});

function TabbedComponents({ fetchData, clearData, chartData }) {
  const navigation = useContext(NavigationContext);

  useEffect(() => {
    fetchData(navigation.getParam('detailTitle'));
    return () => {
      clearData();
    };
  }, [fetchData]);

  return (
    <Tabs style={styles.tabs} tabBarUnderlineStyle={styles.tabBarUnderlineStyle}>
      <Tab
        heading="BUY"
        key="1"
        tabStyle={styles.tabStyle}
        textStyle={styles.textStyle}
        activeTabStyle={styles.activeTabStyle}
        activeTextStyle={styles.textStyle}
      >
        <BuyContainer chartData={chartData} />
      </Tab>

      <Tab
        heading="SELL"
        key="sell"
        tabStyle={styles.tabStyle}
        textStyle={styles.textStyle}
        activeTabStyle={styles.activeTabStyle}
        activeTextStyle={styles.textStyle}
      >
        <SellContainer chartData={chartData} />
      </Tab>
    </Tabs>
  );
}

TabbedComponents.defaultProps = {
  fetchData: null,
  clearData: null,
  chartData: []
};

TabbedComponents.propTypes = {
  fetchData: PropTypes.func,
  clearData: PropTypes.func,
  chartData: presetProps.chartDataTypes
};

const mapStateToProps = ({ instrumentDetail }) => ({
  chartData: instrumentDetail.chartData
});

const mapDispatchToProps = dispatch => ({
  fetchData: symbol => dispatch(thunks.getInstrumentDetailData(symbol)),
  clearData: () => dispatch(actions.clearDetailData())
});

export default connect(mapStateToProps, mapDispatchToProps)(TabbedComponents);
