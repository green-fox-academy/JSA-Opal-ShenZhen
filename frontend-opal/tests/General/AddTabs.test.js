import React from 'react';
import renderer from 'react-test-renderer';

import TabbedComponents from 'components/common/TabbedComponents';

function TestComponent() {
  return <></>;
}

describe('<TabbedComponents />', () => {
  it('renders correctly', () => {
    TestComponent.title = 'test';
    const tree = renderer
      .create(<TabbedComponents components={[TestComponent, TestComponent]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
