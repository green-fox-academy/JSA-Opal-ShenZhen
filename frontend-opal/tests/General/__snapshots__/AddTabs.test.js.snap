// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabbedComponents /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "zIndex": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#a7a6ab",
          "elevation": 3,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around",
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        },
        Object {
          "backgroundColor": "#F8F8F8",
        },
        Object {},
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        active={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F8F8F8",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "rgb(250,110,59)",
              "borderBottomWidth": 4,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "600",
                "marginLeft": 7,
                "marginRight": 7,
              },
              Object {
                "fontSize": 15,
              },
              Object {
                "color": "rgb(250,110,59)",
              },
              Object {
                "color": "rgb(250,110,59)",
              },
            ]
          }
          uppercase={false}
        >
          test
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        active={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F8F8F8",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "white",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 16,
                "marginLeft": 7,
                "marginRight": 7,
              },
              Object {
                "fontSize": 15,
              },
              Object {
                "color": "rgb(250,110,59)",
              },
              Object {
                "color": "rgb(250,110,59)",
              },
            ]
          }
          uppercase={false}
        >
          test
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#007aff",
          "bottom": 0,
          "height": 4,
          "left": 0,
          "position": "absolute",
          "width": 375,
        }
      }
    />
  </View>
  <RCTScrollView
    alwaysBounceVertical={false}
    automaticallyAdjustContentInsets={false}
    contentOffset={
      Object {
        "x": 0,
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="handled"
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={16}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <View
          activeTabStyle={
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "rgb(250,110,59)",
              "borderBottomWidth": 4,
            }
          }
          activeTextStyle={
            Object {
              "color": "rgb(250,110,59)",
            }
          }
          heading="test"
          style={
            Object {
              "backgroundColor": "#FFF",
              "flex": 1,
            }
          }
          tabStyle={
            Object {
              "backgroundColor": "white",
            }
          }
          textStyle={
            Object {
              "color": "rgb(250,110,59)",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <View
          heading="test"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
